packages/uploadthing/src/h3.ts

```diff
@@ -0,0 +1,79 @@
+import type { H3Event } from "h3";
+import {
+  assertMethod,
+  defineEventHandler,
+  getRequestHeaders,
+  getRequestURL,
+  readBody,
+  setHeaders,
+  setResponseStatus,
+} from "h3";
+
+import type { Json } from "@uploadthing/shared";
+import { getStatusCodeFromError, UploadThingError } from "@uploadthing/shared";
+
+import { UPLOADTHING_VERSION } from "./constants";
+import { defaultErrorFormatter } from "./internal/error-formatter";
+import {
+  buildPermissionsInfoHandler,
+  buildRequestHandler,
+} from "./internal/handler";
+import type { RouterWithConfig } from "./internal/handler";
+import type { FileRouter } from "./internal/types";
+import type { CreateBuilderOptions } from "./internal/upload-builder";
+import { createBuilder } from "./internal/upload-builder";
+
+export type { FileRouter } from "./internal/types";
+
+export const createUploadthing = <TErrorShape extends Json>(
+  opts?: CreateBuilderOptions<TErrorShape>,
+) =>
+  createBuilder<
+    { req: undefined; res: undefined; event: H3Event },
+    TErrorShape
+  >(opts);
+
+export const createH3EventHandler = <TRouter extends FileRouter>(
+  opts: RouterWithConfig<TRouter>,
+) => {
+  const requestHandler = buildRequestHandler(opts);
+  const getBuildPerms = buildPermissionsInfoHandler<TRouter>(opts);
+
+  return defineEventHandler(async (event) => {
+    assertMethod(event, ["GET", "POST"]);
+    setHeaders(event, { "x-uploadthing-version": UPLOADTHING_VERSION });
+
+    // GET
+    if (event.method === "GET") {
+      setResponseStatus(event, 200);
```

Comment:
this is default so I think we can avoid....

```suggestion
```

Reply:
Fair enough - the less code the better right